### PR TITLE
Paging/Memory Modularization - Cache/TLB Part

### PR DESCRIPTION
**Pull request by lifeix**
Wrap common APIs for Cache/TLB Part

For Cache, ACRN provides three APIs:
    - flush_invalidate_all_cache
    - flush_cacheline
    - flush_cache_range

For TLB, ACRN provides two APIs:
    - flush_tlb
    - flush_tlb_range

This patch also remove ppt_set/clear_(attribute) to set_paging_(attribute)

Tracked-On: #5830
Signed-off-by: Li Fei1 <fei1.li@intel.com>

### CODE REVIEW

File: hypervisor/arch/x86/cpu.c [hunk 1]
```diff
@@ -446,7 +446,7 @@ void cpu_dead(void)
 		/* clean up native stuff */
 		vmx_off();
 		/* TODO: a cpu dead can't effect the RTVM which use Software SRAM */
```
Comment (dongyaozu) [inline hypervisor/arch/x86/cpu.c:448]: This TODO should be completed now.

File: hypervisor/arch/x86/guest/vmexit.c [hunk 2]
```diff
@@ -409,7 +409,7 @@ static int32_t wbinvd_vmexit_handler(struct acrn_vcpu *vcpu)
 
 	/* GUEST_FLAG_RT has not set in post-launched RTVM before it has been created */
 	if ((!is_software_sram_enabled()) && (!has_rt_vm())) {
-		cache_flush_invalidate_all();
+		wbinvd();
 	} else {
 		if (is_rt_vm(vcpu->vm)) {
 			walk_ept_table(vcpu->vm, ept_flush_leaf_page);
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vmexit.c:415]: This function (to flush all) for both SSRAM case or not, can be a seperate wrap API.

File: hypervisor/arch/x86/cpu.c [hunk 3]
```diff
@@ -446,7 +446,7 @@ void cpu_dead(void)
...
-		cache_flush_invalidate_all();
+		wbinvd();
```
Comment (dongyaozu) [inline hypervisor/arch/x86/cpu.c:449]: why not "flush_invalidate_all_cache"?

Comment (dongyaozu) [reply hypervisor/arch/x86/cpu.c:449]: BTW, given that flush_tlb actually means flush and invalidate, I think we can use flush_cache_all here too.

Comment (lifeix) [reply hypervisor/arch/x86/cpu.c:449]: > why not "flush_invalidate_all_cache"?

IMHO, we should use wbinvd for x86 special code and use flush_invalidate_all_cache for non-arch related code

Comment (lifeix) [reply hypervisor/arch/x86/cpu.c:449]: > BTW, given that flush_tlb actually means flush and invalidate, I think we can use flush_cache_all here too.

Do you mean flush_cache_range(0, all address space) ?

File: hypervisor/arch/x86/guest/vmexit.c (hunk 2 above)
Comment (lifeix) [reply hypervisor/arch/x86/guest/vmexit.c:415]: I think this function would have large logic change when we support multi-RTVM. In that time, we could hide SSRAM from vmexit module.

File: hypervisor/arch/x86/cpu.c (hunk 1 above)
Comment (lifeix) [reply hypervisor/arch/x86/cpu.c:448]: OK.

File: hypervisor/arch/x86/cpu.c (hunk 3 above)
Comment (dongyaozu) [reply hypervisor/arch/x86/cpu.c:449]: Yes

Comment (dongyaozu) [reply hypervisor/arch/x86/cpu.c:449]: x86 code call general code should be fine too.

File: hypervisor/arch/x86/guest/vmexit.c (hunk 2 above)
Comment (dongyaozu) [reply hypervisor/arch/x86/guest/vmexit.c:415]: OK. But call flush_cache_all is better IMO

**Review by dongyaozu — APPROVED** [review]

**Pull request merged**